src/application-insights/azext_applicationinsights/custom.py

```diff
@@ -40,17 +40,46 @@ def get_metrics_metadata(cmd, client, application, resource_group_name=None):
     return client.metrics.get_metadata(get_id_from_azure_resource(cmd.cli_ctx, application, resource_group=resource_group_name))
 
 
-def create_or_update_component(client, application, resource_group_name, location, tags=None, kind="web", application_type='web'):
-    from .vendored_sdks.mgmt_applicationinsights.models import ApplicationInsightsComponent
-    component = ApplicationInsightsComponent(location, kind, application_type=application_type, tags=tags)
+def create_or_update_component(cmd, client, application, resource_group_name, location, tags=None,
+                               kind="web", application_type='web', workspace_resource_id=None,
+                               public_network_access_for_ingestion=None, public_network_access_for_query=None):
+    # due to service limitation, we have to do such a hack. We must refract the logic later.
+    if workspace_resource_id is None:
+        from .vendored_sdks.mgmt_applicationinsights.v2018_05_01_preview.models import ApplicationInsightsComponent
+        from ._client_factory import applicationinsights_mgmt_plane_client
+        client = applicationinsights_mgmt_plane_client(cmd.cli_ctx, api_version='2018-05-01-preview').components
+        component = ApplicationInsightsComponent(location=location, kind=kind, application_type=application_type, tags=tags,
+                                                 public_network_access_for_ingestion=public_network_access_for_ingestion,
+                                                 public_network_access_for_query=public_network_access_for_query)
+        return client.create_or_update(resource_group_name, application, component)
+
+    from .vendored_sdks.mgmt_applicationinsights.v2020_02_02_preview.models import ApplicationInsightsComponent
+    component = ApplicationInsightsComponent(location=location, kind=kind, application_type=application_type,
+                                             tags=tags, workspace_resource_id=workspace_resource_id,
+                                             public_network_access_for_ingestion=public_network_access_for_ingestion,
+                                             public_network_access_for_query=public_network_access_for_query)
     return client.create_or_update(resource_group_name, application, component)
```

Comment:
It's the same client as there is no workspace provided?

Reply:
no. If workspace is not provided, the client would be the default client.